lab_2/main.py

```diff
@@ -46,19 +48,21 @@ def fill_lcs_matrix(first_sentence_tokens: tuple, second_sentence_tokens: tuple)
     :param second_sentence_tokens: a tuple of tokens
     :return: a lcs matrix
     """
-    lcs = []
-    rows = len(first_sentence_tokens)
-    columns = len(second_sentence_tokens)
-    if rows <= 0 and columns <= 0:
+    
+    if not isinstance(first_sentence_tokens, tuple) or not len(first_sentence_tokens) or \
+            not isinstance(second_sentence_tokens, tuple) or not len(second_sentence_tokens) or \
+            not isinstance(first_sentence_tokens[0], str) or not isinstance(second_sentence_tokens[0], str):
         return []
-    if isinstance(first_sentence_tokens, tuple) and isinstance(second_sentence_tokens, tuple):
-        lcs = create_zero_matrix(rows, columns)
-        for ind_1, token_1 in enumerate(first_sentence_tokens):
-            for ind_2, token_2 in enumerate(second_sentence_tokens):
-                if token_1 == token_2:
-                    lcs[ind_1][ind_2] = lcs[ind_1 - 1][ind_2 - 1] + 1
-                else:
-                    lcs[ind_1][ind_2] = max(lcs[ind_1 - 1][ind_2], lcs[ind_1][ind_2 - 1])
+
+    lcs = create_zero_matrix(len(first_sentence_tokens),len(second_sentence_tokens))
+    for ind_1, token_1 in enumerate(first_sentence_tokens):
+        for ind_2, token_2 in enumerate(second_sentence_tokens):
+            if token_1 == token_2:
```

Comment:
still, have a question about ind_1 ==0 and index ==0

Reply:
if ind_1 == 0 or ind_2 == 0, we're on the last row or colomn and have indexes [-1],  other rows or columns become 1 if token_1 == token_2. And than we have another iteration.